hazelcast/src/test/java/testsubjects/NonStaticFunctionFactory.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package testsubjects;
+
+import java.util.function.BiFunction;
+
+public class NonStaticFunctionFactory {
+
+    private NonStaticFunctionFactory() {
+    }
+
+    public static BiFunction<String, String, String> getAnonymousNonSerializableBiFunction(final String returnValue) {
+        return new BiFunction<String, String, String>() {
```

Comment:
Maybe it's an overkill having a separate class for this, you can move this to `ComputeConditionallyTests`. 

Also, can you move the package `testsubjects` under `com.hazelcast.test`. And what does `testsubjects` stand for?

Reply:
**Overall background -** 
In `ComputeConditionallyTests`, ideally I wanted to test scenarios where the client and servers would be on different JVMs, and the client has classes that are NOT present on the server JVM.
To simulate this, I used a custom class-loader for the `server`. That way, the classes visible to the `client` aren't visible to the `server` classloader.

Further to this, I noticed that during class-loading, classes inside `com.hazelcast` have an additional fallback ([link](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/internal/nio/ClassLoaderUtil.java#L244)). I wanted to avoid this fallback for the tests (to make the tests stricter)

**About naming the new package `testsubjects` -** 
I thought we can use this package to host classes that act like "subjects" of tests (classes that aren't tests themselves. Other tests use them like subjects). Happy to rename it ... suggestions?

**About the separate class - **
Yes, it's kind of an overkill. I wanted the anonymous class to have a package other than `com.hazelcast`. Any other solution comes to mind?

Reply:
Forgot about the `com.hazelcast` package rule. Thanks, let's keep it as it is.